lib/linux_admin/network_interface/network_interface_rh.rb

```diff
@@ -0,0 +1,138 @@
+
+    # Set the IPv4 gateway address for this interface
+    #
+    # @param address [String]
+    # @raise ArgumentError if the address is not formatted properly
+    def gateway=(address)
+      validate_ip(address)
+      @interface_conf["GATEWAY"] = address
+    end
+
+    # Set the IPv4 sub-net mask for this interface
+    #
+    # @param mask [String]
+    # @raise ArgumentError if the mask is not formatted properly
+    def netmask=(mask)
+      validate_ip(mask)
+      @interface_conf["NETMASK"] = mask
+    end
+
+    # Sets one or both DNS servers for this network interface
+    #
+    # @param servers Array<String> The DNS servers
+    def dns=(servers)
+      @interface_conf["DNS1"] = servers[0]
+      @interface_conf["DNS2"] = servers[1] if servers[1]
+    end
+
+    # Sets the search domain list for this network interface
+    #
+    # @param domains [Array<String>] the list of search domains
+    def search_order=(domains)
+      @interface_conf["DOMAIN"] = "\"#{domains.join(' ')}\""
+    end
+
+    # Set up the interface to use DHCP
+    # Removes any previously set static networking information
+    def enable_dhcp
+      @interface_conf["BOOTPROTO"] = "dhcp"
+      @interface_conf.delete("IPADDR")
+      @interface_conf.delete("NETMASK")
+      @interface_conf.delete("GATEWAY")
+      @interface_conf.delete("PREFIX")
+      @interface_conf.delete("DNS1")
+      @interface_conf.delete("DNS2")
+      @interface_conf.delete("DOMAIN")
+    end
+
+    # Applies the given static network configuration to the interface
+    #
+    # @param ip [String] IPv4 address
+    # @param mask [String] subnet mask
+    # @param gw [String] gateway address
+    # @param dns1 [Array<String>] list of dns servers
+    # @param search [Array<String>] list of search domains
+    # @return [Boolean] true on success, false otherwise
+    # @raise ArgumentError if an IP is not formatted properly
+    def apply_static(ip, mask, gw, dns, search = nil)
+      self.address      = ip
+      self.netmask      = mask
+      self.gateway      = gw
+      self.dns          = dns
+      self.search_order = search if search
+      save
+    end
+
+    # Writes the contents of @interface_conf to @iface_file as `key`=`value` pairs
+    # and resets the interface
+    #
+    # @return [Boolean] true if the interface was successfully brought up with the
+    #   new configuration, false otherwise
+    def save
+      old_contents = File.read(@iface_file)
+
+      return false unless ifdown
+
+      File.write(@iface_file, @interface_conf.delete_blanks.collect { |k, v| "#{k}=#{v}" }.join("\n"))
+
+      unless ifup
+        File.write(@iface_file, old_contents)
+        ifup
+        return false
+      end
+
+      true
+    end
+
+    private
+
+    # Validate that the given IPv4 address is formatted correctly
+    #
+    # @param ip [String]
+    # @raise ArgumentError if the address is not correctly formatted
+    def validate_ip(ip)
+      IPAddr.new(ip)
+    rescue
```

Comment:
@carbonin Does `IPAddr::InvalidAddressError` inherit from `ArgumentError`?
If not, I think this should at least be `rescue ArgumentError, IPAddr::InvalidAddressError` to prevent rescuing unwanted errors